### PR TITLE
Feature/optional debug toolbar

### PR DESCRIPTION
**Pull request by stuaxo**
Enable debug toolbar if ENABLE_DEBUG_TOOLBAR is set and DEBUG is True.

With tests.

### CODE REVIEW

File: config/tests/test_settings_toolbar.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    assert 'debug_toolbar.middleware.DebugToolbarMiddleware' not in \
+           settings.MIDDLEWARE_CLASSES
+
+
+@toolbar_enabled
+def test_toolbar_app_present(settings):
+    assert 'debug_toolbar' in settings.INSTALLED_APPS
+
+
+@toolbar_enabled
```
Comment (sdonk) [inline config/tests/test_settings_toolbar.py]: you can change the settings in the context of each test and test all the scenarios

Comment (sdonk) [reply config/tests/test_settings_toolbar.py]: `pytest-django' has a fixture called `settings`

Comment (stuaxo) [reply config/tests/test_settings_toolbar.py]: Cool, I was wondering about this.

- So if I change it to set or unset ENABLE_DEBUG_TOOLBAR then I guess the everything else should just work ?

Comment (sdonk) [reply config/tests/test_settings_toolbar.py]: I would write 2 tests: 

- set `ENABLE_DEBUG_TOOLBAR` to `True` and assert that `MIDDLEWARE_CLASSES`, `INSTALLED_APPS` and `INTERNAL_IPS` are set correctly
- set `ENABLE_DEBUG_TOOLBAR` to `False` and  assert that `MIDDLEWARE_CLASSES`, `INSTALLED_APPS` and `INTERNAL_IPS` don't contain the debut toolbar stuff

Comment (stuaxo) [reply config/tests/test_settings_toolbar.py]: Are there any existing projects using it ?   I'm having a bit of trouble finding an example.

Comment (stuaxo) [reply config/tests/test_settings_toolbar.py]: Why my last comment is after yours, I'll have a look at doing that.

Comment (stuaxo) [reply config/tests/test_settings_toolbar.py]: Will definitely catch up tomorrow.    

Obviously setting ENABLE_DEBUG_TOOLBAR should set the other settings, but then that means the settings themselves should be reloaded, which is where I got stuck + I'm quite new to pytest :)

Comment (stuaxo) [reply config/tests/test_settings_toolbar.py]: Re: our conversation I'm going to merge as is, as the other options are tricky to implement and settings aren't important enough.

**Pull request merged**